src/components/Project/ProjectNew.js

```diff
@@ -0,0 +1,300 @@
+import {Link} from 'react-router-dom';
+import {withRouter} from 'react-router';
+import Select from 'react-select';
+
+const styles = {
+  dates: {
+    marginTop: '15px',
+  },
+  participants: {
+    marginBottom: '20px',
+    padding: '0px',
+  },
+  clients: {
+    padding: '0px',
+    margin: '0px 0px 20px 0px',
+  },
+  summary: {
+    height: '400px !important',
+  },
+  control: {
+    control: (base, state) => ({
+      ...base,
+      borderColor: 'lightgrey',
+      boxShadow: `0 0 0 1px lightgrey`,
+      borderRadius: '4px',
+      '&:hover': {
+        borderColor: '#1fbfd7',
+        boxShadow: `1 1 5 10px #1fbfd7`,
+        color: 'white',
+      },
+    }),
+  },
+};
+
+class AddNewProject extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      clientSelect: '',
+      client_id: '',
+      title: '',
+      summary: '',
+      start_date: '',
+      end_date: '',
+      participantSelect: [],
+      participants: [],
+      clients: [],
+      users: [],
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleClientChange = this.handleClientChange.bind(this);
+    this.handleParticipantChange = this.handleParticipantChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('https://lesewert.herokuapp.com/api/v1/users')
+      .then(res => res.json())
+      .then(data =>
+        this.setState({
+          users: data.users,
+        }),
+      )
+      .catch(() => console.log('error'))
+      .then(
+        fetch('https://lesewert.herokuapp.com/api/v1/clients')
+          .then(res => res.json())
+          .then(data =>
+            this.setState({
+              clients: data.clients,
+            }),
+          )
+          .catch(() => console.log('error')),
+      );
+  }
+
+  handleChange(e) {
+    e.preventDefault();
+    this.setState({
+      [e.target.name]: e.target.value,
+    });
+  }
+
+  handleClientChange(option) {
+    this.setState({
+      clientSelect: option,
+    });
+  }
+
+  handleParticipantChange(option) {
+    this.setState({
+      participantSelect: option,
+    });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.clientSelect !== this.state.clientSelect) {
+      this.setState({
+        client_id: this.state.clientSelect.value,
+      });
+    } else if (prevState.participantSelect !== this.state.participantSelect) {
+      this.setState({
+        participants: [
+          ...this.state.participants,
+          this.state.participantSelect[this.state.participantSelect.length - 1]
+            .value,
+        ],
+      });
+    }
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    fetch('/api/v1/projects', {
+      method: 'POST',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+      }),
+      body: JSON.stringify(this.state),
+    })
+      .then(res => {
+        if (res.status >= 200 && res.status < 300) {
```

Comment:
do we have to do that? i have never seen this before, also when i google it i dont see anyone else doing that

Reply:
Yes as I only want to route back to the projects table if the data is sent and correct. If it throws an error the user can fix what went wrong before the state is reset and all their inputs are lost